mlflow/store/file_store.py

```diff
@@ -410,19 +419,28 @@ def _get_run_files(self, run_uuid, resource_type):
         if run_info is None:
             raise MlflowException("Run '%s' metadata is in invalid state." % run_uuid,
                                   databricks_pb2.INVALID_STATE)
+
+        _, run_dir = self._find_run_root(run_uuid)
+        # run_dir exists since run validity has been confirmed above.
+        return self._get_metric_files(run_dir, resource_type)
+
+    def _get_experiment_files(self, experiment_id):
+        _validate_experiment_id(experiment_id)
+        experiment_dir = self._get_experiment_path(experiment_id, assert_exists=True)
+        return self._get_metric_files(experiment_dir, "tag")
```

Comment:
The only non-generic portion of the resource function is the subfolder_name - I think it doesn't make too much sense to copy the subsequent logic to both functions separately just because it happens to be for runs as well as experiments. I did refactor some of the resource type logic out, so hopefully that should resolve the concerns about logical isolation.